### DISCUSSION TITLE
DeskPi Pro USB

### DISCUSSION THREAD

**Discussion by GLink** [bareboat-necessities/lysmarine_gen#338] [Q&A]
Hi, anyone having experience with the DeskPi Pro case with Lysmarine? Have followed all the guidelines, but the extra front USB ports still not working and therefore fan is not working either. Is there anything interfering with the dcw2 overlay?

Thanks
Gaute

**Comment by mgrouch** [comment]
For fan control you need to configure which usb port is used on your system. 
Default is /dev/ttyUSB0 which might be not the case for you. 

see notes in

https://github.com/DeskPi-Team/deskpi/tree/master/drivers

Generally it’s documented 
here https://github.com/DeskPi-Team/deskpi

Front USB ports should work after applying these procedures. They do for me but I might have different revision of the board than you. 

If it still doesn’t work open issue on 

https://github.com/DeskPi-Team/deskpi

directly

**Reply by GLink** [reply]
Thanks, but I think this is more related to the Lysmarine build, because when I deploy the stock Raspberry 64 bit Bullseye build on the same HW, the front USB and fan control is working as expected. I suspect the following setting in /boot/config.txt,  dtoverlay=dwc2,dr_mode=host does not take effect on the Lysmarine build, but I'm not good enough in Linux to figure out why.   /dev/ttyUSB0 is not present after boot and the front USB ports is not working (tried to move the USB cable for the mouse from back to front)

**Comment by mgrouch** [comment]
You need to add that line at the end 
of /boot/config.txt file. 

BBN OS has it only in CM4 section. Which is for industrial raspberry pi.

**Reply by GLink** [reply]
Yes, has it on the end of config.txt in the [All] section, just as on the plain Raspberry build, but it is not taking effect, which is the problem.

**Reply by mgrouch** [reply]
It surely works for me. I just tried on my DeskPi pro. Can you post here your whole config.txt?

**Reply by GLink** [reply]
Thanks for testing, but now I'm more confused: Now Telegraf (daemon) no longer start, but USB is now working (and thereby also the fan). I need to debug more...

**Reply by mgrouch** [reply]
You probably updated telegraf and it needs some simple configuration if you are using it. If you do not use it you can simply disable its service.
